packages/next/build/babel/preset.ts

```diff
@@ -70,12 +70,16 @@ module.exports = (api: any, options: NextBabelPresetOptions = {}): BabelPreset =
         // @babel/plugin-transform-react-jsx-self automatically in development
         development: isDevelopment || isTest,
         ...options['preset-react']
-      }]
+      }],
+      require('@babel/preset-typescript')
```

Comment:
This looks very wrong tbh, it should only apply for .ts / .tsx

Reply:
The preset already handles this more proper than we already were. Only the plugin requires the overrides section.